NetCoreMentoring/src/NetCoreMentoring.App/Infrastructure/ImageCacheMiddleware.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using NetCoreMentoring.Core;
+using NetCoreMentoring.Core.Utilities;
+
+namespace NetCoreMentoring.App.Infrastructure
+{
+    // I use caching with help of attribute instead
+    // Don't delete this just for completing homework module
+    public class ImageCacheMiddleware
+    {
+        private readonly RequestDelegate next;
+        private static readonly Dictionary<string, string> SupportedPictureContentTypes = new Dictionary<string, string>()
+        {
+            {"image/jpeg", "jpeg"},
+            {"image/png", "png"}
+        };
+        private readonly string CacheImagePath;
+        private readonly IConfiguration _configuration;
+
+        public ImageCacheMiddleware(RequestDelegate next, IConfiguration configuration)
```

Comment:
Consider adding a new options class and passing it here:

    public class CachingOptions
    {
        public Dictionary<string, string> CachedCodePath { get;set; }
    }
    // todo: add the rest of caching related options to here

...

    public ImageCacheMiddleware(RequestDelegate next, IConfiguration configuration, CachingOptions cachingOptions)
    {
        ...
    }

Reply:
done